### PR TITLE
Fix true colors generation for AHI HSD data and refactor the dep tree code

### PR DESCRIPTION
**Pull request by mraspaud**
Refactor the dep tree code and fix the true color composite for instruments with channels at different resolutions.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
The idea is that if the compositor `C` needs dependency `D1` with property `P`, and `D1` depends on `D2`, then `D2` should have property `P` also (if applicable). Indeed, this isn't strict transitivity, but I couldn't come up with a better name for it. Suggestions are welcome.

File: satpy/node.py [hunk 1]
```diff
@@ -17,16 +17,22 @@
...
 EMPTY_LEAF_NAME = "__EMPTY_LEAF_SENTINEL__"
 
 
+class MissingDependencies(RuntimeError):
+    """Exception when dependencies are missing."""
+
+    def __init__(self, missing_dependencies, *args, **kwargs):
+        """Set up the exception."""
+        super().__init__(*args, **kwargs)
+        self.missing_dependencies = missing_dependencies
```
Comment (djhoese) [inline satpy/node.py:33]: Can/should this have a custom __repr__ or __str__ to provide the typical missing dependency error message?

Comment (mraspaud) [reply satpy/node.py:33]: Good idea

Comment (mraspaud) [reply satpy/node.py:33]: Fixed

**Comment by mraspaud** [comment]
I think things are in order with public/private

**Pull request merged**